apps/bucket/cloudformation.yml

```
@@ -150,16 +154,18 @@ Resources:
           - s3:List*
           - s3:Get*
           Effect: Allow
-          Principal: "*"
+          Principal:
+            - "*"
+          Resource:
+          - !Sub arn:aws:s3:::${DataSetBucket}/*
+          - !Sub arn:aws:s3:::${DataSetBucket}
+        Statement:
+        - Action:
+          - s3:Put*
+          Effect: Allow
+          Principal:
+            - !Sub arn:aws:iam::${AwsApplicationAccountId}:${FetcherRoleName}
           Resource:
           - !Sub arn:aws:s3:::${DataSetBucket}/*
           - !Sub arn:aws:s3:::${DataSetBucket}
```

Specifying `s3:PutObject` for the action should be sufficient.

No need to include `arn:aws:s3:::${DataSetBucket}` (the bucket) in the resources; just `arn:aws:s3:::${DataSetBucket}/*` (objects in the bucket).

Syntactically equivalent, but I prefer to use `key: value` rather than a list with a single element for IAM policies. Looks like the AWS-delivered template errs the other way, though, so feel free to pick either.

```suggestion
        Statement:
        - Action: s3:PutObject
          Effect: Allow
          Principal: !Sub arn:aws:iam::${AwsApplicationAccountId}:${FetcherRoleName}
          Resource:!Sub arn:aws:s3:::${DataSetBucket}/*
```